Sources/Afluent/SequenceOperators/RetryAfterFlatMappingSequence.swift

```diff
@@ -134,6 +134,11 @@ extension AsyncSequences {
             } catch {
                 guard !(error is CancellationError) else { throw error }
 
+                guard let unwrappedError = (error as? Failure),
+                      unwrappedError == state.error else {
```

Comment:
This pattern is repeated a lot, I wonder if we should just define a pattern matching function for matching an `any Error` to some specific error.

Reply:
This ended up being a neat little refactor!